client.go

```diff
@@ -29,6 +68,92 @@ type RestClient struct {
 	baseUrl url.URL
 }
 
+func NewNetworkClient(config NetworkConfig) (client *Client, err error) {
```

Comment:
does it make sense to split this into 'new for template' for devnet/testnet/mainnet and 'new for URLs' ?
Maybe 'NewNetworkClient' should be 'NewClient(netname)', and maybe 'NewCustomClient(url,url,url,url)' ?

Reply:
I'm not really in tune with what is "go like"

I suspect more configuration will have to be added here... like optional timeouts, API keys etc.

This provides the flexibility to add in future versions extra information that allows for that.  Either that or we use a builder pattern, where we allow users to add more things as they go